include/seqan/index/find_index_lambda.h

```diff
@@ -122,17 +122,17 @@ _findBacktracking(TIndexIt indexIt,
             {
                 do
                 {
-                    // Mismatch.
-                    TThreshold delta = !ordEqual(parentEdgeLabel(indexIt), value(needleIt));
-                    _findBacktracking(indexIt, needle, needleIt + 1,
-                                      static_cast<TThreshold>(errors + delta), threshold, delegate, TDistance());
-
                     // Deletion.
                     if (IsSameType<TDistance, EditDistance>::VALUE)
                     {
                         _findBacktracking(indexIt, needle, needleIt,
                                           static_cast<TThreshold>(errors + 1), threshold, delegate, TDistance());
                     }
+
+                    // Mismatch.
+                    TThreshold delta = !ordEqual(parentEdgeLabel(indexIt), value(needleIt));
+                    _findBacktracking(indexIt, needle, needleIt + 1,
+                                      static_cast<TThreshold>(errors + delta), threshold, delegate, TDistance());
                 }
```

Comment:
This only influences the order of the results, right?
Would you know by change, @esiragusa? 

Or you, @cpockrandt ?